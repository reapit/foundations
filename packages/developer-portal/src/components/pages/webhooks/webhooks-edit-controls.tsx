import { Button, ButtonGroup, FlexContainer } from '@reapit/elements'
import React, { Dispatch, FC, SetStateAction } from 'react'
import { WebhookModel } from '../../../services/webhooks'
import { WebhooksPingForm } from './webhook-ping-form'
import { ExpandableContentType } from './webhooks-manage'
import { WebhooksManageForm } from './webhooks-manage-form'

interface WebhooksEditControlsProps {
  webhookModel: WebhookModel
  expandableContentType: ExpandableContentType
  setExpandableContentType: Dispatch<SetStateAction<ExpandableContentType>>
  setIndexExpandedRow: Dispatch<SetStateAction<number | null>>
}

export const handleSetContentType =
  (
    expandableContentType: ExpandableContentType,
    setExpandableContentType: Dispatch<SetStateAction<ExpandableContentType>>,
  ) =>
  () => {
    setExpandableContentType(expandableContentType)
  }

export const WebhooksEditControls: FC<WebhooksEditControlsProps> = ({
  webhookModel,
  expandableContentType,
  setExpandableContentType,
  setIndexExpandedRow,
}) => (
  <>
    {expandableContentType === ExpandableContentType.Controls && (
      <FlexContainer isFlexJustifyCenter>
        <ButtonGroup>
          <Button
            intent="primary"
            onClick={handleSetContentType(ExpandableContentType.Manage, setExpandableContentType)}
          >
            Edit
          </Button>
          <Button intent="primary" onClick={handleSetContentType(ExpandableContentType.Ping, setExpandableContentType)}>
            Ping
          </Button>
        </ButtonGroup>
      </FlexContainer>
    )}
    {expandableContentType === ExpandableContentType.Ping && (
      <WebhooksPingForm
        webhookModel={webhookModel}
        setIndexExpandedRow={setIndexExpandedRow}
        setExpandableContentType={setExpandableContentType}
      />
    )}
    {expandableContentType === ExpandableContentType.Manage && (
      <WebhooksManageForm
        webhookModel={webhookModel}
        setIndexExpandedRow={setIndexExpandedRow}
        setExpandableContentType={setExpandableContentType}
      />
    )}
  </>
)
