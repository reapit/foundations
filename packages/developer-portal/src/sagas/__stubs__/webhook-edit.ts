import {
  WebhookSubscription,
  WebhookModal,
  WebhookEditState,
  WebhookCreateEditState,
} from '@/reducers/webhooks-subscriptions/webhook-edit-modal'
import { CreateWebhookParams } from '@/actions/webhooks-subscriptions'
import { installationsStub } from './installations'

export const webhookDataStub: WebhookSubscription = {
  subcriptionCustomers: {
    ...installationsStub,
  },
  subcriptionTopics: {
    pageCount: 0,
    pageNumber: 1,
    pageSize: 25,
    totalCount: 0,
    _embedded: [
      {
        id: 'SOME_ID',
        name: 'SOME_NAME',
        description: 'SOME_DESCRIPTION',
        url: 'https://example.com',
        active: true,
      },
    ],
  },
}

export const webhookItemDataStub: WebhookModal = {
  id: 'SOME_ID',
  applicationId: 'SOME_APP_ID',
  url: 'https://example.com',
  description: 'SOME_DESCRIPTION',
  topicIds: ['SOME_ID'],
  customerIds: ['SOME_ID'],
  ignoreEtagOnlyChanges: true,
  active: false,
}
export const webhookEditDataStub: WebhookEditState = {
  ...webhookDataStub,
  webhookData: webhookItemDataStub,
  loading: false,
  modalType: '',
  webhookCreateEditState: WebhookCreateEditState.INITIAL,
}

export const createWebhookItem: CreateWebhookParams = {
  applicationId: '',
  url: '',
  description: '',
  topicIds: [],
  customerIds: [],
  ignoreEtagOnlyChanges: true,
  active: true,
}
