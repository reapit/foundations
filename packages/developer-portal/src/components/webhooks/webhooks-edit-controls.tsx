import { Button, ButtonGroup, FlexContainer } from '@reapit/elements'
import { GetActionNames, getActions } from '@reapit/use-reapit-data'
import { useReapitGet } from '@reapit/use-reapit-data'
import React, { Dispatch, FC, SetStateAction, useState } from 'react'
import CopyToClipboard from 'react-copy-to-clipboard'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { WebhookModel, WebhookPublicKeyResponse } from '../../types/webhooks'
import { WebhooksPingForm } from './webhook-ping-form'
import { ExpandableContentType } from './webhooks-manage'
import { WebhooksManageForm } from './webhooks-manage-form'

interface WebhooksEditControlsProps {
  webhookModel: WebhookModel
  expandableContentType: ExpandableContentType
  setExpandableContentType: Dispatch<SetStateAction<ExpandableContentType>>
  setIndexExpandedRow: Dispatch<SetStateAction<number | null>>
  refreshSubscriptions: () => void
}

export const handleSetContentType =
  (
    expandableContentType: ExpandableContentType,
    setExpandableContentType: Dispatch<SetStateAction<ExpandableContentType>>,
  ) =>
  () => {
    setExpandableContentType(expandableContentType)
  }

export const handleCopyPublicKey = (setCopyState: Dispatch<SetStateAction<boolean>>) => () => {
  setCopyState(true)

  setTimeout(() => {
    setCopyState(false)
  }, 5000)
}

export const handleFetchKey = (setFetchKey: Dispatch<SetStateAction<boolean>>) => () => {
  setFetchKey(true)
}

export const WebhooksEditControls: FC<WebhooksEditControlsProps> = ({
  webhookModel,
  expandableContentType,
  setExpandableContentType,
  setIndexExpandedRow,
  refreshSubscriptions,
}) => {
  const [copyState, setCopyState] = useState<boolean>(false)
  const [fetchKey, setFetchKey] = useState<boolean>(false)

  const [publicKeyResponse, fetchingPublicKey] = useReapitGet<WebhookPublicKeyResponse>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getPublicWebhookKey],
    queryParams: {
      applicationId: webhookModel.applicationId,
    },
    fetchWhenTrue: [webhookModel.applicationId, fetchKey],
  })

  return (
    <>
      {expandableContentType === ExpandableContentType.Controls && (
        <FlexContainer isFlexJustifyCenter>
          <ButtonGroup>
            <Button
              intent="primary"
              onClick={handleSetContentType(ExpandableContentType.Manage, setExpandableContentType)}
            >
              Edit
            </Button>
            <Button
              intent="primary"
              onClick={handleSetContentType(ExpandableContentType.Ping, setExpandableContentType)}
            >
              Ping
            </Button>
            {publicKeyResponse && publicKeyResponse.keys ? (
              <CopyToClipboard
                text={JSON.stringify(publicKeyResponse.keys[0])}
                onCopy={handleCopyPublicKey(setCopyState)}
              >
                <Button intent="default">{copyState ? 'Public Key Copied' : 'Copy Public Key'}</Button>
              </CopyToClipboard>
            ) : (
              <Button
                intent="primary"
                loading={fetchingPublicKey}
                disabled={fetchingPublicKey}
                onClick={handleFetchKey(setFetchKey)}
              >
                Fetch Public Key
              </Button>
            )}
          </ButtonGroup>
        </FlexContainer>
      )}
      {expandableContentType === ExpandableContentType.Ping && (
        <WebhooksPingForm
          webhookModel={webhookModel}
          setIndexExpandedRow={setIndexExpandedRow}
          setExpandableContentType={setExpandableContentType}
        />
      )}
      {expandableContentType === ExpandableContentType.Manage && (
        <WebhooksManageForm
          webhookModel={webhookModel}
          setIndexExpandedRow={setIndexExpandedRow}
          setExpandableContentType={setExpandableContentType}
          refreshSubscriptions={refreshSubscriptions}
        />
      )}
    </>
  )
}
