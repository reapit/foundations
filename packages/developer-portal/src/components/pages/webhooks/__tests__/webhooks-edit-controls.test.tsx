import React from 'react'
import { handleSetContentType, WebhooksEditControls } from '../webhooks-edit-controls'
import { webhookItemDataStub } from '../../../../sagas/__stubs__/webhook-edit'
import { render } from '../../../../tests/react-testing'

describe('WebhooksEditControls', () => {
  it('should match a snapshot with small content', () => {
    expect(
      render(
        <WebhooksEditControls
          webhookModel={webhookItemDataStub}
          expandableContentSize="small"
          setExpandableContentSize={jest.fn()}
          setIndexExpandedRow={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('should match a snapshot with medium content', () => {
    expect(
      render(
        <WebhooksEditControls
          webhookModel={webhookItemDataStub}
          expandableContentSize="medium"
          setExpandableContentSize={jest.fn()}
          setIndexExpandedRow={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('should match a snapshot with large content', () => {
    expect(
      render(
        <WebhooksEditControls
          webhookModel={webhookItemDataStub}
          expandableContentSize="large"
          setExpandableContentSize={jest.fn()}
          setIndexExpandedRow={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })
})

describe('handleSetContentType', () => {
  it('should set expandable content size', () => {
    const expandableContentSize = 'large'
    const setExpandableContentSize = jest.fn()
    const curried = handleSetContentType(expandableContentSize, setExpandableContentSize)
    curried()
    expect(setExpandableContentSize).toHaveBeenCalledWith('large')
  })
})
