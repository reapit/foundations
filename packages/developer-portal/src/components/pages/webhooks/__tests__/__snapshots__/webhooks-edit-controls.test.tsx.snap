// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhooksEditControls should match a snapshot with controls content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ecbqyq6-el-snack-holder"
      />
      <div
        class="e1de2pym-el-flex-justify-center e2kmzjs-el-flex-container"
      >
        <div
          class="e19a1osy-el-button-group"
        >
          <div
            class="eay05y4-el-button-group-inner"
          >
            <button
              class="ebee6r1-el-intent-primary e1f2ikqj-el-button"
            >
              Edit
            </button>
            <button
              class="ebee6r1-el-intent-primary e1f2ikqj-el-button"
            >
              Ping
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ecbqyq6-el-snack-holder"
    />
    <div
      class="e1de2pym-el-flex-justify-center e2kmzjs-el-flex-container"
    >
      <div
        class="e19a1osy-el-button-group"
      >
        <div
          class="eay05y4-el-button-group-inner"
        >
          <button
            class="ebee6r1-el-intent-primary e1f2ikqj-el-button"
          >
            Edit
          </button>
          <button
            class="ebee6r1-el-intent-primary e1f2ikqj-el-button"
          >
            Ping
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WebhooksEditControls should match a snapshot with manage content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ecbqyq6-el-snack-holder"
      />
      <form
        class="el6khyg-el-p8"
      >
        <div
          class="e1fwulnz-el-form-layout"
        >
          <div
            class="ee8eqmp-el-input-wrap-full"
          >
            <div
              class="e1q0zwo9-el-flex-justify-between ef457sc-el-flex-align-center e2kmzjs-el-flex-container"
            >
              <h2
                class="ek1d8t2-el-ml3 eh96l6x-el-has-bold-text e2dm5rh-el-subtitle"
              >
                Edit Webhook
              </h2>
              <div
                class="e19a1osy-el-button-group"
              >
                <div
                  class="e1i2en9o-el-button-group-align-right eay05y4-el-button-group-inner"
                >
                  <button
                    class="e1ymtwpu-el-intent-low e1f2ikqj-el-button"
                    type="button"
                  >
                    Delete
                  </button>
                  <button
                    class="epibvz0-el-intent-secondary e1f2ikqj-el-button"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="ebee6r1-el-intent-primary e1tqqdf9-el-button-has-right-chevron e1f2ikqj-el-button"
                    type="submit"
                  >
                    Update
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ee8eqmp-el-input-wrap-full"
          >
            <div
              class="e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                name="url"
                placeholder="Enter secure https:// url"
              />
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                Webhook URL
              </label>
            </div>
          </div>
          <div
            class="ee8eqmp-el-input-wrap-full"
          >
            <div
              class="e1x57i2-el-flex-wrap e1q0zwo9-el-flex-justify-between ef457sc-el-flex-align-center ehvkknk-el-mb5 e2kmzjs-el-flex-container"
            >
              <div
                class="s18kkr5u-search-min-width e1979rmu-el-input-group"
              >
                <input
                  class="e1urjlvd-el-input"
                  id="test-static-id"
                  placeholder="Search topics to get started"
                />
                <span
                  class="ej9yrfo-el-icon"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M19.7556 18.5783L14.0682 12.8909C15.1699 11.5301 15.8332 9.80089 15.8332 7.91758C15.8332 3.55262 12.2815 0.000976562 7.91656 0.000976562C3.55161 0.000976562 0 3.55258 0 7.91754C0 12.2825 3.55165 15.8342 7.9166 15.8342C9.7999 15.8342 11.5291 15.1708 12.8899 14.0692L18.5773 19.7566C18.7398 19.9191 18.9531 20.0008 19.1665 20.0008C19.3798 20.0008 19.5932 19.9191 19.7557 19.7566C20.0815 19.4308 20.0815 18.9041 19.7556 18.5783ZM7.9166 14.1675C4.46996 14.1675 1.66666 11.3642 1.66666 7.91754C1.66666 4.4709 4.46996 1.6676 7.9166 1.6676C11.3632 1.6676 14.1665 4.4709 14.1665 7.91754C14.1665 11.3642 11.3632 14.1675 7.9166 14.1675Z"
                        fill="currentColor"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0"
                      >
                        <rect
                          fill="white"
                          height="20.0003"
                          width="20"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </span>
                <label
                  class="e1nbkkrq-el-label"
                  for="test-static-id"
                >
                  Subscription topics
                </label>
              </div>
            </div>
            <div
              class="eooz7jy-el-multi-select-input-wrapper"
            >
              <input
                class="eg4njbl-el-multi-select-input"
                id="topic-edit-ids-SOME_ID"
                name="topicIds"
              />
              <div
                class="e12c6sou-el-multi-select-selected"
              >
                <input
                  checked=""
                  class="e1wdkc9j-el-multi-select-checkbox"
                  id="test-static-id"
                  type="checkbox"
                />
                <label
                  class="e1x92nc7-el-multi-select-label"
                  for="test-static-id"
                >
                  SOME_NAME
                </label>
              </div>
            </div>
          </div>
          <div
            class="ee8eqmp-el-input-wrap-full"
          >
            <label
              class="ek1d8t2-el-ml3 ehvkknk-el-mb5 e1nbkkrq-el-label"
            >
              Subscription customers
            </label>
            <div
              class="eooz7jy-el-multi-select-input-wrapper"
            >
              <input
                class="eg4njbl-el-multi-select-input"
                id="customer-edit-ids-SOME_ID"
                name="customerIds"
              />
              <div
                class="e12c6sou-el-multi-select-selected"
              >
                <input
                  checked=""
                  class="e1wdkc9j-el-multi-select-checkbox"
                  id="test-static-id"
                  type="checkbox"
                />
                <label
                  class="e1x92nc7-el-multi-select-label"
                  for="test-static-id"
                >
                  SOME_NAME
                </label>
              </div>
              <div
                class="ef7odf0-el-multi-select-un-selected"
              >
                <input
                  class="e1wdkc9j-el-multi-select-checkbox"
                  id="test-static-id"
                  type="checkbox"
                />
                <label
                  class="ehgp5gp-el-has-grey-chips e1x92nc7-el-multi-select-label"
                  for="test-static-id"
                >
                  All Customers
                </label>
                <input
                  class="e1wdkc9j-el-multi-select-checkbox"
                  id="test-static-id"
                  type="checkbox"
                />
                <label
                  class="ehgp5gp-el-has-grey-chips e1x92nc7-el-multi-select-label"
                  for="test-static-id"
                >
                  Sandbox Estates
                </label>
              </div>
            </div>
          </div>
          <div
            class="ee6e7by-el-input-wrap"
          >
            <label
              class="e1nbkkrq-el-label"
            >
              Status
            </label>
            <input
              class="e1ih6lkl-el-toggle-checkbox"
              id="status-edit-toggle-SOME_ID"
              name="active"
              type="checkbox"
            />
            <label
              class="ehi76vp-el-has-grey-bg e1ott14k-el-toggle-label"
              for="status-edit-toggle-SOME_ID"
            >
              <span
                class="e18omfqv-el-toggle-item"
              >
                Active
              </span>
              <span
                class="e18omfqv-el-toggle-item"
              >
                Inactive
              </span>
            </label>
          </div>
          <div
            class="ee6e7by-el-input-wrap"
          >
            <div
              class="e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                name="ignoreEtagOnlyChanges"
                type="checkbox"
              />
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                Ignore where only the etag has been modified
              </label>
              <span
                class="e15fmnse-el-input-add-on"
              >
                Ignore
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="ecbqyq6-el-snack-holder"
    />
    <form
      class="el6khyg-el-p8"
    >
      <div
        class="e1fwulnz-el-form-layout"
      >
        <div
          class="ee8eqmp-el-input-wrap-full"
        >
          <div
            class="e1q0zwo9-el-flex-justify-between ef457sc-el-flex-align-center e2kmzjs-el-flex-container"
          >
            <h2
              class="ek1d8t2-el-ml3 eh96l6x-el-has-bold-text e2dm5rh-el-subtitle"
            >
              Edit Webhook
            </h2>
            <div
              class="e19a1osy-el-button-group"
            >
              <div
                class="e1i2en9o-el-button-group-align-right eay05y4-el-button-group-inner"
              >
                <button
                  class="e1ymtwpu-el-intent-low e1f2ikqj-el-button"
                  type="button"
                >
                  Delete
                </button>
                <button
                  class="epibvz0-el-intent-secondary e1f2ikqj-el-button"
                  type="button"
                >
                  Cancel
                </button>
                <button
                  class="ebee6r1-el-intent-primary e1tqqdf9-el-button-has-right-chevron e1f2ikqj-el-button"
                  type="submit"
                >
                  Update
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ee8eqmp-el-input-wrap-full"
        >
          <div
            class="e1979rmu-el-input-group"
          >
            <input
              class="e1urjlvd-el-input"
              id="test-static-id"
              name="url"
              placeholder="Enter secure https:// url"
            />
            <label
              class="e1nbkkrq-el-label"
              for="test-static-id"
            >
              Webhook URL
            </label>
          </div>
        </div>
        <div
          class="ee8eqmp-el-input-wrap-full"
        >
          <div
            class="e1x57i2-el-flex-wrap e1q0zwo9-el-flex-justify-between ef457sc-el-flex-align-center ehvkknk-el-mb5 e2kmzjs-el-flex-container"
          >
            <div
              class="s18kkr5u-search-min-width e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                placeholder="Search topics to get started"
              />
              <span
                class="ej9yrfo-el-icon"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M19.7556 18.5783L14.0682 12.8909C15.1699 11.5301 15.8332 9.80089 15.8332 7.91758C15.8332 3.55262 12.2815 0.000976562 7.91656 0.000976562C3.55161 0.000976562 0 3.55258 0 7.91754C0 12.2825 3.55165 15.8342 7.9166 15.8342C9.7999 15.8342 11.5291 15.1708 12.8899 14.0692L18.5773 19.7566C18.7398 19.9191 18.9531 20.0008 19.1665 20.0008C19.3798 20.0008 19.5932 19.9191 19.7557 19.7566C20.0815 19.4308 20.0815 18.9041 19.7556 18.5783ZM7.9166 14.1675C4.46996 14.1675 1.66666 11.3642 1.66666 7.91754C1.66666 4.4709 4.46996 1.6676 7.9166 1.6676C11.3632 1.6676 14.1665 4.4709 14.1665 7.91754C14.1665 11.3642 11.3632 14.1675 7.9166 14.1675Z"
                      fill="currentColor"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0"
                    >
                      <rect
                        fill="white"
                        height="20.0003"
                        width="20"
                      />
                    </clippath>
                  </defs>
                </svg>
              </span>
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                Subscription topics
              </label>
            </div>
          </div>
          <div
            class="eooz7jy-el-multi-select-input-wrapper"
          >
            <input
              class="eg4njbl-el-multi-select-input"
              id="topic-edit-ids-SOME_ID"
              name="topicIds"
            />
            <div
              class="e12c6sou-el-multi-select-selected"
            >
              <input
                checked=""
                class="e1wdkc9j-el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="e1x92nc7-el-multi-select-label"
                for="test-static-id"
              >
                SOME_NAME
              </label>
            </div>
          </div>
        </div>
        <div
          class="ee8eqmp-el-input-wrap-full"
        >
          <label
            class="ek1d8t2-el-ml3 ehvkknk-el-mb5 e1nbkkrq-el-label"
          >
            Subscription customers
          </label>
          <div
            class="eooz7jy-el-multi-select-input-wrapper"
          >
            <input
              class="eg4njbl-el-multi-select-input"
              id="customer-edit-ids-SOME_ID"
              name="customerIds"
            />
            <div
              class="e12c6sou-el-multi-select-selected"
            >
              <input
                checked=""
                class="e1wdkc9j-el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="e1x92nc7-el-multi-select-label"
                for="test-static-id"
              >
                SOME_NAME
              </label>
            </div>
            <div
              class="ef7odf0-el-multi-select-un-selected"
            >
              <input
                class="e1wdkc9j-el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="ehgp5gp-el-has-grey-chips e1x92nc7-el-multi-select-label"
                for="test-static-id"
              >
                All Customers
              </label>
              <input
                class="e1wdkc9j-el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="ehgp5gp-el-has-grey-chips e1x92nc7-el-multi-select-label"
                for="test-static-id"
              >
                Sandbox Estates
              </label>
            </div>
          </div>
        </div>
        <div
          class="ee6e7by-el-input-wrap"
        >
          <label
            class="e1nbkkrq-el-label"
          >
            Status
          </label>
          <input
            class="e1ih6lkl-el-toggle-checkbox"
            id="status-edit-toggle-SOME_ID"
            name="active"
            type="checkbox"
          />
          <label
            class="ehi76vp-el-has-grey-bg e1ott14k-el-toggle-label"
            for="status-edit-toggle-SOME_ID"
          >
            <span
              class="e18omfqv-el-toggle-item"
            >
              Active
            </span>
            <span
              class="e18omfqv-el-toggle-item"
            >
              Inactive
            </span>
          </label>
        </div>
        <div
          class="ee6e7by-el-input-wrap"
        >
          <div
            class="e1979rmu-el-input-group"
          >
            <input
              class="e1urjlvd-el-input"
              id="test-static-id"
              name="ignoreEtagOnlyChanges"
              type="checkbox"
            />
            <label
              class="e1nbkkrq-el-label"
              for="test-static-id"
            >
              Ignore where only the etag has been modified
            </label>
            <span
              class="e15fmnse-el-input-add-on"
            >
              Ignore
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WebhooksEditControls should match a snapshot with ping content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ecbqyq6-el-snack-holder"
      />
      <form
        class="el6khyg-el-p8"
      >
        <h2
          class="ek1d8t2-el-ml3 e2dm5rh-el-subtitle"
        >
          Test Webhook Subscription
        </h2>
        <div
          class="e20pb60-el-grid"
        >
          <div
            class="eb3z2eu-el-col-split"
          >
            <label
              class="ek1d8t2-el-ml3 e1nbkkrq-el-label"
            >
              Subscription topics
            </label>
            <div
              class="e1979rmu-el-input-group"
            >
              <select
                class="e60u3kb-el-select"
                name="topicId"
              >
                <option
                  value=""
                >
                  None selected
                </option>
                <option
                  value="SOME_ID"
                >
                  SOME_NAME
                </option>
              </select>
            </div>
          </div>
          <div
            class="eb3z2eu-el-col-split"
          >
            <p
              class="evblxhy-el-body-text"
            >
              To test your Webhook subscription, please select a subscription topic and an example payload for that topic will be sent to the configured URL. For more information, please
               
              <a
                class="l3gjl5x-link"
              >
                click here
              </a>
            </p>
            <div
              class="et0k7gk-el-ml-auto e19a1osy-el-button-group"
            >
              <div
                class="eay05y4-el-button-group-inner"
              >
                <button
                  class="epibvz0-el-intent-secondary e1f2ikqj-el-button"
                  type="button"
                >
                  Cancel
                </button>
                <button
                  class="ebee6r1-el-intent-primary e1tqqdf9-el-button-has-right-chevron e1f2ikqj-el-button"
                  type="submit"
                >
                  Test
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="ecbqyq6-el-snack-holder"
    />
    <form
      class="el6khyg-el-p8"
    >
      <h2
        class="ek1d8t2-el-ml3 e2dm5rh-el-subtitle"
      >
        Test Webhook Subscription
      </h2>
      <div
        class="e20pb60-el-grid"
      >
        <div
          class="eb3z2eu-el-col-split"
        >
          <label
            class="ek1d8t2-el-ml3 e1nbkkrq-el-label"
          >
            Subscription topics
          </label>
          <div
            class="e1979rmu-el-input-group"
          >
            <select
              class="e60u3kb-el-select"
              name="topicId"
            >
              <option
                value=""
              >
                None selected
              </option>
              <option
                value="SOME_ID"
              >
                SOME_NAME
              </option>
            </select>
          </div>
        </div>
        <div
          class="eb3z2eu-el-col-split"
        >
          <p
            class="evblxhy-el-body-text"
          >
            To test your Webhook subscription, please select a subscription topic and an example payload for that topic will be sent to the configured URL. For more information, please
             
            <a
              class="l3gjl5x-link"
            >
              click here
            </a>
          </p>
          <div
            class="et0k7gk-el-ml-auto e19a1osy-el-button-group"
          >
            <div
              class="eay05y4-el-button-group-inner"
            >
              <button
                class="epibvz0-el-intent-secondary e1f2ikqj-el-button"
                type="button"
              >
                Cancel
              </button>
              <button
                class="ebee6r1-el-intent-primary e1tqqdf9-el-button-has-right-chevron e1f2ikqj-el-button"
                type="submit"
              >
                Test
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
